### PR TITLE
validate network_id

### PR DESCRIPTION
**Pull request by naved001**
This PR is to fix issue #584 
As suggested by @gsilvis, I added a method (validate_network_id()) to the NetworkAllocator class to check the validity of the net_id. 
What are your thoughts on this @shwsun ?

### CODE REVIEW

**Comment by gsilvis** [comment]
Something weird is going on in the postgres testsuite... your tests are apparently /hanging/.

**Comment by shwsun** [comment]
Looks sth is breaking, and you may want to test it on a real postgres env first.

**Comment by naved001** [comment]
now it works.

File: haas/ext/network_allocators/vlan_pool.py [hunk 1]
```diff
@@ -63,6 +63,15 @@ def is_legal_channel_for(self, channel_id, net_id):
     def get_default_channel(self):
         return "vlan/native"
 
+    def validate_network_id(self, net_id):
+        try:
+            if 1 <= int(net_id) <= 4096:
+                return True
+            else:
+                return False
```
Comment (gsilvis) [inline haas/ext/network_allocators/vlan_pool.py]: Suggest:  ``return 1 <= int(ned_id) <= 4096``

**Comment by gsilvis** [comment]
Everything else looks good to me.  (I forgot about the 'review' functionality while making my comments...)

**Comment by naved001** [comment]
okay. I'll make the changes you suggested.

**Comment by gsilvis** [comment]
Note to whoever merges:  please remember to 'Squash and merge'

**Comment by gsilvis** [comment]
@shwsun @knikolla @henn Can we get a second review here?

**Comment by gsilvis** [comment]
@zenhack You might be ineterested specifically, because this affects the Network Allocator api

**Pull request merged**